desc/utils.py

```diff
@@ -431,6 +431,41 @@ def svd_inv_null(A):
     return Ainv, Z
 
 
+def qr_xp_null(A, b, tol=1e-10):
+    """Compute null space of a matrix and particular solution Ax=b using QR.
+
+    Parameters
+    ----------
+    A : ndarray
+        Matrix to find null space of.
+    b : ndarray
+        Right-hand side of Ax = b.
+
+    Returns
+    -------
+    x_p : ndarray
+        Particular solution to Ax = b.
+    Z : ndarray
+        Null space of A such that AZ=0 and Z.T@Z=I.
+
+    """
+    # Linear constraint matrix A is usually wide
+    # QR decomposition of A^T
+    Q, R = qr(A.T)
```

Comment:
we can check this can't we? in case it ends up not being wide?

though I guess if it is not wide that means it is overdetermined

Reply:
Yes, if it is tall then we have overdetermined and I don't think in DESC you can create that scenario by existing constraints, and since we clean degenerate stuff and fixed constraints already, we remove a lot of rows. So, I believe it is impossible to get tall A here, but I can add a check if there is a possibility.